src/System.IO.FileSystem/src/System/IO/UnixFileSystemObject.cs

```diff
@@ -20,13 +20,15 @@ internal sealed class UnixFileSystemObject : IFileSystemObject
             /// <summary>The last cached stat information about the file.</summary>
             private Interop.libcoreclr.fileinfo _fileinfo;
 
-            /// <summary>
-            /// Whether we've successfully cached a stat structure.
-            /// -1 if we need to refresh _fileinfo, 0 if we've successfully cached one,
-            /// or any other value that serves as an errno error code from the
-            /// last time we tried and failed to refresh _fileinfo.
-            /// </summary>
-            private int _fileinfoInitialized = -1;
+            enum InitializationState : byte
+            {
+                RefreshNeeded,     // we need to refresh _fileInfo
+                SuccessfullyCached, // we've successfully cached the stat structure
+                ErrorEncountered   // we hit an error on last try, get errno from _InitializatonErrno 
+            }
+
+            private InitializationState _fileinfoInitialization;
+            private Interop.ErrorInfo _fileinfoInitializationError; 
```

Comment:
The byte will fit in prior padding next to existing bool, I think. But ErrorInfo is 2x errno. I opted to put the code back and reconvert the Error in the case where we throw. Seems like the right tradeoff as we get the size win for always and pay a slight time penalty only in the error case.